### DISCUSSION TITLE
Templater stopped returning data.

### DISCUSSION THREAD

**Discussion by ghost** [SilentVoid13/Templater#824] [Help]
Templater seems to have stopped working for me. 

```
[[<% tp.date.yesterday("YYYY-MM-DD") %>]] [[<% tp.date.tomorrow("YYYY-MM-DD") %>]]

<% tp.web.daily_quote() %>

```
These render as is rather than returning the date(s) and quote. I've tried creating a new vault with Templater and these do not return data there either. These snippets have worked in the past and I haven't altered them since then. What could cause this? How do I troubleshoot it? 

Void Linux
Templater version 1.14.1
Obsidian 0.15.9

**Comment by shabegom** [comment]
1. Always check the dev console for an error. 
2. A common problem is a user script with an error. If you have any user scripts try removing them and see if it works.

**Reply by ghost** [reply]
Nothing relevant in the console as far as I can tell. No mention of it loading templater but perhaps that is to be expected. Removing all scripts did not make a difference. As I said the issue occurs in a new vault with nothing but templater installed (no scripts; nothing). 

![fsdaf](https://user-images.githubusercontent.com/102994494/189659657-11611530-77b6-4205-8294-0075653cdd8f.png)

**Comment by ZeroDayOwl** [comment]
Having the same problem, only `Last Modified` seems to render for me. The rest are not rendered, printing only the raw commands in the notes.

I have tried with a fresh install of the plugin, and cross checking each command once again. But it just doesn't work at all.

**Reply by shabegom** [reply]
Post the full template

**Reply by danraymond** [reply]
I too have the same issue and I have not altered any templates for many months??

**Reply by shabegom** [reply]
Quite possible the latest update has affected your templates, but without more information it is impossible to troubleshoot.

**Reply by danraymond** [reply]
Thank you, maybe update to 16 did it
For example this no longer works in my YAML Header;

---
creation date: <% tp.file.creation_date() %>
modification date: <% tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>
---

**Reply by shabegom** [reply]
Any errors in the dev console?

What other templater commands are in the note?

Any user script functions?

Are you using any of the following anywhere in your vault:

tp.file.path
to.file.exists

**Comment by xxxELGUAPOxxx** [comment]
I have exactly the same problem after latest plugin update (1.14.2). 
I am on Obsidian v0.15.9
Any ideas how to fix / downgrade plugin?

**Reply by AB1908** [reply]
You can use BRAT to install a specific version/tag. You can also install it by hand by taking the contents of the release and dumping it into the plugin's folder.

**Comment by xxxELGUAPOxxx** [comment]
Thanks for your reply @AB1908 
Could you please tell me where I can get the previous version of the Templater plugin (1.14.1)?
I am heavy Obsidian user, but new on Github so your help is much appreciated.
Cheers!

**Reply by AB1908** [reply]
https://github.com/SilentVoid13/Templater/releases

**Reply by xxxELGUAPOxxx** [reply]
Downgrade worked, many thanks! Looks like there is a bug in 1.14.2.

**Comment by shabegom** [comment]
If anyone is willing to get on a zoom call to troubleshoot this issue, please reach out on the OKG Discord templater thread. Thanks!

**Reply by xxxELGUAPOxxx** [reply]
Hi @shabegom 

Not able to join the call RN, but here is what i found so far:

Yesterday I updated the templater plugin to 1.14.2 and since then my daily templates stopped working (the templater parts of my page remain unprocessed).

Example of code I use on my daily note pages:

---
date: <% tp.date.now("YYYY-MM-DD") %>
---

or this as a page header 
# <% tp.date.now("dddd, MMMM D, YYYY",0, tp.file.title) %>

These parts of newly created daily notes are ignored by last version of templater. After downgrade to 1.14.1 everything works fine again. 

I am on Obsidian v0.15.9 and Windows 11.

Hope this helps!

**Comment by shabegom** [comment]
[main.js.zip](https://github.com/SilentVoid13/Templater/files/9600032/main.js.zip)

Can someone who is experiencing this problem unzip the main.js file and put it in $YOUR_VAULT/.obsidian/plugins/templater-obsidian
reload the vault and try running something that wasn't working before.

report back if this fixes it.

**Reply by xxxELGUAPOxxx** [reply]
Yes, that worked!
Upgraded back to 1.14.2 - broken again
Then replaced  the provided main.js file and everything seems to be working.

**Comment by shabegom** [comment]
Please update to 1.14.3

**Reply by xxxELGUAPOxxx** [reply]
Wow, that was fast. Thanks!

**Reply by ZeroDayOwl** [reply]
Hey thanks for it! 

Also, I discovered something, putting a `+` after the starting bracket sort of fixed it to work while it's Dynamically rendered in Live Preview mode. 

Like this, 
```
<%+ tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %> 
```
Instead of,
```
<% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %> 
```

**Reply by danraymond** [reply]
This works for me also (on 1.14.3) I downgraded to 1.14.1following this thread but that didn't fix mine, the + however still works (but not in the YAML.

**Reply by ZeroDayOwl** [reply]
Yeah, it just renders, but doesn't actually replace the function with the value.

**Comment by ZeroDayOwl** [comment]
@shabegom I upgraded to Templater v.1.14.3, but without any luck. The commands don't render on my system, until I specifically put a `+` to dynamically render them, unlike before. I also replaced the main.js file in the plugin-dir, but it just doesn't seem to work for me.

The commands I'm trying to render - 
```
<% tp.date.now("YYYYMMDDHHmm") %>
<% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %> 
<%+ tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>
```

I'm using Obsidian v0.15.9
Templater v1.14.3
Windows 10

**Reply by AB1908** [reply]
What do you mean by render? Templater doesn't really render anything beyond the dynamic command stuff.

**Reply by shabegom** [reply]
Can you provide more info on what you're trying to do and what isn't working? Be more specific than you think you need to be. It will help narrow the problem space.

**Reply by ZeroDayOwl** [reply]
@AB1908 Yeah, I meant by that. When I placed `+` at the beginning of the command, it considers it as a [Dynamic command](https://silentvoid13.github.io/Templater/commands/dynamic-command.html) and renders it in Live Preview mode.

**Reply by ZeroDayOwl** [reply]
@shabegom Sure. I have those commands in the top of the template for my notes: 

```
<% tp.date.now("YYYYMMDDHHmm") %>
Creation date : <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %> 
Last Modified : <%+ tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>
Path           : <% tp.file.path(true) %>
```
Usually before the functions got executed, returning the values in the notes. But after that recent update, it stopped working. Even if I invoke the template in a new note, only the `Last Modified` gets rendered in the Live Preview mode (since it's a dynamic command) and others stay the same they were.

```
<% tp.date.now("YYYYMMDDHHmm") %>
Creation date  : <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
Last Modified  : Tuesday 20st September 2022 12:54:20
Path           : <% tp.file.path(true) %>
```

**Reply by xxxELGUAPOxxx** [reply]
@danraymond @ZeroDayOwl 
I got into a similar issue before and then I realised that was my fault because I used shortcut for the core template plugin. The core plugin will not trigger the render so you have to use either  "<%" templater icon in taskbar or define a new shortcut for this plugin. Anyway 1.14.2 broke the plugin but 1.14.3 works fine again for me.

**Reply by shabegom** [reply]
@ZeroDayOwl if you then run the "replace templates in current note" command, do those turn into dates, or do they stay as templater commands.

**Reply by xxxELGUAPOxxx** [reply]
Hi guys, I did some experiments and can confirm the issue. Current version Templater 1.14.3 is probably colliding with Dataview plugin.
If you put templater functions in Dataview's Frontmatter or YAML or whatewer you call it (simply between the 3 dashes at tge top of the page, lol). It will report METADATA (INVALID YAML) - the error shows only in READ MODE. When I put the same functions outside the dashes, everything works fine without any errors in both modes.

**Reply by xxxELGUAPOxxx** [reply]
> @ZeroDayOwl if you then run the "replace templates in current note" command, do those turn into dates, or do they stay as templater commands.

When I run "replace templates in current note" in the template with templater commands in YAML, it will turn into dates. But when I switch to READ MODE there is the INVALID YAML error.

**Reply by ZeroDayOwl** [reply]
Hey @shabegom sorry I couldn't reply earlier.

If I run the `Replace Templates in current note` command, they turn into dates yeah. But I have the `Trigger Templater on New file creation` option on, so shouldn't it just replace the functions to their values automatically like before instead of having to run the command separately?

**Comment by xxxELGUAPOxxx** [comment]
UPDATE: Looks like you can't put  ```<% tp.date.now("YYYYMMDDHHmm") %>``` into YAML alone. There must be some word with colon before or it will cause the YAML error after render. Also dynamic commands will not be executed when in YAML. Hope this helps!

EXAMPLE:

This one will give you YAML error and the Last modified will not work
```
---
<% tp.date.now("YYYYMMDDHHmm") %>
Creation date  : <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
Path                 : <% tp.file.path(true) %>
Last modified  : <%+ tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>
---
```

This one will work OK
 ```
---
ID: <% tp.date.now("YYYYMMDDHHmm") %>
Creation date  : <% tp.file.creation_date("dddd Do MMMM YYYY HH:mm:ss") %>
Path                 : <% tp.file.path(true) %>
---
Last modified  : <%+ tp.file.last_modified_date("dddd Do MMMM YYYY HH:mm:ss") %>

```

**Reply by danraymond** [reply]
I am on 1.14.3 and the rendered one works but I still get nothing from the data in the YAML using either example. Even with DataView disabled. So not sure what's going on yet

**Reply by AB1908** [reply]
Sorry for duplicating efforts but what was the issue again?

**Comment by shabegom** [comment]
Move all further discussion to issue #836. Thanks
